networkx/generators/classic.py

```diff
@@ -386,6 +386,114 @@ def grid_graph(dim,periodic=False):
     H=nx.relabel_nodes(G, flatten)
     H.name="grid_graph(%s)"%dlabel
     return H
+ 
+def triangle_lattice(rows, cols, sidelength, offset = (0,0), sig_dig=2, periodic=False, create_using=None):
+    """Generates a triangle lattice with rows rows and cols columns.
+    Each triangle has edges of length sidelength.
+    All triangles either point up or down.
+    A row is a horizontal row of triangles that point up and point down - their center points are in a line
+    A column is a vertical column of triangles that point up and point down - again their center points are in a line.    
+    
+    The node names are given by their coordinates as tuples, rounded to sig_dig decimal places.
+    This can be offset in space with offset=(xshift,yshift).
+
+    written by Joel Miller"""
```

Comment:
You can add yourself to `__author__` or `credits.rst` if you wish. I don't think this is the right place for writing the name of author.